### ISSUE TITLE
Should this library move to yaml.v3

### ISSUE THREAD

**Issue by errordeveloper** [kubernetes-sigs/yaml#18]
[`gopkg.in/go-yaml.v3`](https://github.com/go-yaml/yaml) is already available, should we upgrade to it?

**Comment by dims** [comment]
yes, go for it! note that we would have to trial run changes using a WIP PR in k/k

**Comment by neolit123** [comment]
the API calls in v3 changed, so if this is done we might have to fork.

see the discussion in this PR:
https://github.com/kubernetes-sigs/yaml/pull/26

**Comment by nitrocode** [comment]
This library https://github.com/go-yaml/yaml is now archived as of Apr 1, 2025

It might be better to switch to https://github.com/goccy/go-yaml

**Comment by ivankatliarchuk** [comment]
Is there is a reason why go-yaml instead of goccy? Just trying to understand, as quite few libraries depends on this project?
